src/github/webhook.rs

```diff
@@ -0,0 +1,357 @@
+
+            log::info!("handling issue comment {:?}", payload);
+
+            Event::IssueComment(payload)
+        }
+        EventName::Issue | EventName::PullRequest => {
+            let mut payload = deserialize_payload::<IssuesEvent>(&payload)
+                .context(format!("{:?} failed to deserialize", event))
+                .map_err(anyhow::Error::from)?;
+
+            if matches!(event, EventName::PullRequest) {
+                payload.issue.pull_request = Some(PullRequestDetails::new());
+            }
+
+            log::info!("handling issue event {:?}", payload);
+
+            Event::Issue(payload)
+        }
+        EventName::Push => {
+            let payload = deserialize_payload::<PushEvent>(&payload)
+                .with_context(|| format!("{:?} failed to deserialize", event))
+                .map_err(anyhow::Error::from)?;
+
+            log::info!("handling push event {:?}", payload);
+
+            Event::Push(payload)
+        }
+        EventName::Create => {
+            let payload = deserialize_payload::<CreateEvent>(&payload)
+                .with_context(|| format!("{:?} failed to deserialize", event))
+                .map_err(anyhow::Error::from)?;
+
+            log::info!("handling create event {:?}", payload);
+
+            Event::Create(payload)
+        }
+        // Other events need not be handled
+        EventName::Other => {
+            return Ok(false);
+        }
+    };
+    let errors = crate::handlers::handle(&ctx, &event).await;
+    let mut other_error = false;
+    let mut message = String::new();
+    for err in errors {
+        match err {
+            HandlerError::Message(msg) => {
+                if !message.is_empty() {
+                    message.push_str("\n\n");
+                }
+                message.push_str(&msg);
+            }
+            HandlerError::Other(err) => {
+                log::error!("handling event failed: {:?}", err);
+                other_error = true;
+            }
+        }
+    }
+    if !message.is_empty() {
+        if let Some(issue) = event.issue() {
+            let cmnt = ErrorComment::new(issue, message);
+            cmnt.post(&ctx.github).await?;
+        } else {
+            log::error!("handling event failed: {:?}", message);
+        }
+    }
+    if other_error {
+        Err(WebhookError(anyhow::anyhow!(
+            "handling failed, error logged",
+        )))
+    } else {
+        Ok(true)
+    }
+}
+
+#[derive(Debug)]
+pub struct SignedPayloadError;
+
+impl fmt::Display for SignedPayloadError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "failed to validate payload")
+    }
+}
+
+impl std::error::Error for SignedPayloadError {}
+
+pub fn check_payload_signed(signature: &str, payload: &[u8]) -> Result<(), SignedPayloadError> {
+    let signature = signature
+        .strip_prefix("sha256=")
+        .ok_or(SignedPayloadError)?;
+    let signature = match hex::decode(&signature) {
```

Comment:
`let Ok(signature) = .. else {}`.

Reply:
We use the `Err` variant, so can't use let-else here.